crates/sled/src/tree.rs

```diff
@@ -819,19 +820,56 @@ impl Tree {
     /// assert_eq!(r.next().unwrap(), Ok((vec![5], IVec::from(vec![50]))));
     /// assert_eq!(r.next(), None);
     ///
-    /// let mut r = t.scan(&[2]).rev();
+    /// let mut r = t.scan(&[3]).rev();
+    /// assert_eq!(r.next().unwrap(), Ok((vec![5], IVec::from(vec![50]))));
+    /// assert_eq!(r.next().unwrap(), Ok((vec![4], IVec::from(vec![40]))));
+    /// assert_eq!(r.next().unwrap(), Ok((vec![3], IVec::from(vec![30]))));
+    /// assert_eq!(r.next(), None);
+    /// ```
+    pub fn scan<K>(&self, key: K) -> Iter
+    where
+        K: AsRef<[u8]>,
+    {
+        let key = key.as_ref().to_vec();
```

Comment:
Cloning the just passed byte slice reference value given by the user to make it be an `Vec` doesn't really follow the [C-CALLER-CONTROL flexibility point of the api guideline](https://rust-lang-nursery.github.io/api-guidelines/flexibility.html#c-caller-control).

Reply:
But I understand that you want to expose a simple and flexible API, I agree with your choice. I was just wondering if there was no other solution (i.e. `Into<Vec<u8>>`).